### PR TITLE
Feature: generating restaurant result from responses

### PR DESCRIPTION
**Pull request by TeeWhyKay**
## Why

This pull request is needed because:

Random restaurants are returned previously. Now an algorithm is created to take in the inputs (budget, cuisine, and location) and give the highest rated restaurant according to these inputs.

[Link to Card](https://trello.com/c/WG0Mshms)

## What

This pull request introduces the following:

 * Responses are now generated based on lowest budget amongst group, geolocation center between response locations, and majority cuisine choice amongst the responses.
 * If there's a cuisine tie, randomly sample amongst the tied cuisines
 * If there's no restaurant within 5km of geographic center, remove the location filtering.
 * If there's no results from all 3 inputs, randomly select.
 * Seeds updated to get description and body

## How to Test

1. Create session
2. Generate fake responses
3. Generate result as chomp session organiser

### CODE REVIEW

File: app/controllers/chomp_sessions_controller.rb [hunk 1]
```diff
@@ -62,4 +62,59 @@ def chomp_params
...
+    lowest_budget = responses.minimum('budget')
+    restaurant_pricing = determine_pricing(lowest_budget)
+
+    cuisine_arr = []
+    responses.each do |response|
+      response_cuisine = response.cuisine
+      cuisine_arr << response_cuisine.slice!(1, response_cuisine.length) unless response_cuisine == [""]
+    end
+    cuisine_arr.flatten!(1)
+    tallied_result = cuisine_arr.tally
```
Comment (mxng) [inline app/controllers/chomp_sessions_controller.rb:92]: Wow didin't know about .tally!

File: app/controllers/chomp_sessions_controller.rb [hunk 2]
```diff
@@ -62,4 +62,59 @@ def chomp_params
...
+    most_frequent_cuisine_kv = tallied_result.max_by { |_, value| value }
```
Comment (mxng) [inline app/controllers/chomp_sessions_controller.rb:93]: Wooow didn't know this

File: app/controllers/chomp_sessions_controller.rb [hunk 3]
```diff
@@ -62,4 +62,59 @@ def chomp_params
...
+
+    # check tie
+    tied_results = tallied_result.select { |_, v| v == most_frequent_cuisine_kv[1] }
```
Comment (mxng) [inline app/controllers/chomp_sessions_controller.rb:96]: What does the [1] mean? The key-value pair with key "1"? Just trying to understand.

**Review by mxng — COMMENTED** [review]
Thanks so much for adding this! Seems fine code-wise but I haven't tried it out yet. Will do in class.

File: app/controllers/chomp_sessions_controller.rb (hunk 3 above)
Comment (TeeWhyKay) [reply app/controllers/chomp_sessions_controller.rb:96]: https://ruby-doc.org/core-2.7.3/Enumerable.html#method-i-max_by
Tallied result is a hash containing key value pairs where keys are the cuisine and value is the frequency.
Max_by method returns an **array** containing key in the zeroth index and value on the first index regardless of the fact that there's one or more maximum. Eg. ["Chinese", 5] if there are 5 votes for chinese after 5 responses.

So I have to manually check if there is a tie, by accessing the value at index 1. I think the _kv threw you off. But it's for me know that the data structure is an array but the first value is a key(cuisine) second value is value(frequency).

**Review by mxng — APPROVED** [review]
Works fine!

**Pull request merged**